Source/DataSources/PolygonGeometryUpdater.js

```diff
@@ -216,17 +222,30 @@ define([
         if (positions.length === 0) {
             return;
         }
+        var ellipsoid = this._scene.mapProjection.ellipsoid;
 
-        var centroid = Cartesian3.clone(Cartesian3.ZERO, result);
-        var length = positions.length;
-        for (var i = 0; i < length; i++) {
-            centroid = Cartesian3.add(positions[i], centroid, centroid);
-        }
-        centroid = Cartesian3.multiplyByScalar(centroid, 1 / length, centroid);
-        if (defined(this._scene.globe)) {
-            centroid = this._scene.globe.ellipsoid.scaleToGeodeticSurface(centroid, centroid);
+        var tangentPlane = EllipsoidTangentPlane.fromPoints(positions, ellipsoid);
+        var positions2D = tangentPlane.projectPointsOntoPlane(positions, scratch2DPositions);
+
+        var length = positions2D.length;
+        var area = 0;
+        var j = length - 1;
+        var centroid2D = new Cartesian2();
+        for (var i = 0; i < length; j = i++) {
+            var p1 = positions2D[i];
+            var p2 = positions2D[j];
+            var f = p1.x * p2.y - p2.x * p1.y;
+
+            var sum = Cartesian2.add(p1, p2, cart2Scratch);
+            sum = Cartesian2.multiplyByScalar(sum, f, sum);
+            centroid2D = Cartesian2.add(centroid2D, sum, centroid2D);
+
+            area += f;
         }
-        return centroid;
+
+        var a = 1 / (area * 6);
```

Comment:
Shouldn't this be `var a = 1.0 / (area * 3.0);`?

Reply:
Yes, you're right. Thanks